drivers/can/can_mcan.c

```diff
@@ -386,8 +388,8 @@ int can_mcan_init(const struct device *dev, const struct can_mcan_config *cfg,
 	}
 
 	/* No memset because only aligned ptr are allowed */
-	for (uint32_t *ptr = (uint32_t *)msg_ram;
-	     ptr < (uint32_t *)msg_ram +
+	for (uint32_t *ptr = (u32 *)msg_ram;
+	     ptr < (u32 *)msg_ram +
```

Comment:
Could this be fixed by #41273 instead of adding `__may_alias`?

Reply:
Yep. I commented in the pr you referring to.

Reply:
Yes, but you still reply on code that uses `__may_alias`, so it doesn't remove the usage.